src/domain/prompt/AdvancedSelectOnePrompt.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Flow Interoperability Project (flowinterop.org)
+ * Flow Runner
+ * Copyright (c) 2019, 2020 Viamo Inc.
+ * Authored by: Brett Zabos (brett.zabos@viamo.io)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ **/
+
+import {BasePrompt, IAdvancedSelectOne, IAdvancedSelectOnePromptConfig, ValidationException} from '../..'
+
+export const ADVANCED_SELECT_ONE_PROMPT_KEY = 'AdvancedSelectOne'
+
+export class AdvancedSelectOnePrompt extends BasePrompt<IAdvancedSelectOnePromptConfig> {
+  validate(selectedRow?: IAdvancedSelectOne[]): boolean {
+    const {choiceRows} = this.config
+
+    selectedRow?.forEach(selection => {
+      if (choiceRows.find(row => row.includes(selection.name)) === null)
+        throw new ValidationException('Value provided must be in list of choices')
+    })
+
+    return true
```

Comment:
Question... Can this ever return false? Should the error that is thrown actually return false?

Reply:
No it can't. But, tbh here, I'm just following the pattern in all the other prompts which either throw an error or return false. @bzabos could probably provide insight as to why it is done this way.

According to the FlowRunner class as part of the run() method call stack this line of code eventually handles all the prompts that expect input

```
try {
      prompt.validate(prompt.value)
      return false
    } catch (e) {
      return true
    } 
```

Reply:
On another note, the `selection` has both `name` (matching one of the header cells) and `value` (corresponding to one of the actual row values). And I believe we must use both - to check if the row is valid. Something like:
```
const {choiceRows, choiceRowFields, isResponseRequired} = this.config

if (isResponseRequired) {
  const hasSelectedRow = choiceRows.some(row => {
    selectedRow?.every(selection => {
      const columnIndex = choiceRowFields.indexOf(selection.name)
      if (columnIndex < 0) {
        throw new ValidationException(`Failed to find a column called: ${selection.name}`)
      } else {
        return selection.value === row[columnIndex]
      }
    })
  })
  if (!hasSelectedRow) {
    throw new ValidationException(`Failed to find the given row: ${selectedRow}`)
  }
}

return true
```
I have a similar code in the clipboard's `AdvancedSelectOneComponent.getRowIndexFromDirectorySelect()` method

Reply:
Just noticed a bug. Here's a fixed version (one extra pair of {} is removed):

```
const {choiceRows, choiceRowFields, isResponseRequired} = this.config

if (isResponseRequired) {
  const hasSelectedRow = choiceRows.some(row =>
    selectedRow?.every(selection => {
      const columnIndex = choiceRowFields.indexOf(selection.name)
      if (columnIndex < 0) {
        throw new ValidationException(`Failed to find a column called: ${selection.name}`)
      } else {
        return selection.value === row[columnIndex]
      }
    })
  )
  if (!hasSelectedRow) {
    throw new ValidationException(`Failed to find the given row: ${selectedRow}`)
  }
}

return true
```

Reply:
Appreciate the fix. Updated the branch with it.

Reply:
No problem 😊